mobile/src/main/java/org/openhab/habdroid/core/connection/AbstractConnection.java

```diff
@@ -97,24 +98,60 @@ public String getOpenHABUrl() {
     }
 
     @Override
-    public boolean checkReachabilityInBackground() {
+    public boolean checkReachability() {
         Log.d(TAG, "Checking reachability of " + getOpenHABUrl());
+
+        Socket s;
         try {
-            URL url = new URL(getOpenHABUrl());
-            int checkPort = url.getPort();
-            if (url.getProtocol().equals("http") && checkPort == -1)
-                checkPort = 80;
-            if (url.getProtocol().equals("https") && checkPort == -1)
-                checkPort = 443;
-            Socket s = new Socket();
-            s.connect(new InetSocketAddress(url.getHost(), checkPort), 1000);
-            Log.d(TAG, "Socket connected");
-            s.close();
-            return true;
-        } catch (Exception e) {
-            Log.e(TAG, e.getMessage());
+            s = createConnectedSocket();
+        } catch (MalformedURLException e) {
+            Log.d(TAG, "URL could not be parsed", e);
             return false;
         }
+
+        if (s == null) {
+            return false;
+        }
+
+        Log.d(TAG, "Socket connected");
+        try {
+            s.close();
+        } catch (IOException e) {
+            Log.d(TAG, "Could not close socket connection.", e);
+        }
+        return true;
+    }
+
+    private Socket createConnectedSocket() throws MalformedURLException {
+        for (int retries = 0; retries < MAX_CONNECT_TRIES; retries++) {
+            Socket s = new Socket();
+            InetSocketAddress socketAddress = getReachabilitySocketAddress();
+            try {
+                s.connect(socketAddress, 1000);
```

Comment:
In case of socket timeout, this now waits up to 12 seconds (10 x 1000ms timeout + 10 x 200ms sleep). I think that's a bit too much.